tests/src/test/330+/scala/com/nvidia/spark/rapids/ParquetFieldIdSuite.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids
+
+import java.io.File
+
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.execution.datasources.parquet.ParquetUtils
+import org.apache.spark.sql.types.{IntegerType, MetadataBuilder, StructType}
+
+/**
+ * TODO should update after cuDF supports field id
+ * See https://github.com/NVIDIA/spark-rapids/issues/4846
+ */
+class ParquetFieldIdSuite extends SparkQueryCompareTestSuite {
+
+  // this should failed
+  test("try to write field id") {
+    val tmpFile = File.createTempFile("field-id", ".parquet")
+    try {
+      def withId(id: Int) =
+        new MetadataBuilder().putLong(ParquetUtils.FIELD_ID_METADATA_KEY, id).build()
+      // not support writing field id
+      val schema = new StructType().add("c1", IntegerType, nullable = true, withId(1))
+      val data = (1 to 4).map(i => Row(i))
+
+      assertThrows[IllegalArgumentException] {
```

Comment:
This is not how we normally test for CPU fallbacks.  `IllegalArgumentException` could catch all sorts of exceptions that aren't actually the one we want/expect and then the test passes for the wrong reasons.  Typically we capture the plan and examine it to verify certain nodes are present and/or others are not present.

Also we prefer to put the fallback tests in the Python integration tests which have some nice framework to handle fallback testing.  See `assert_gpu_fallback_write` and the other fallback assertions in asserts.py.

As for the Parquet field ID metadata, you should be able to construct that in pyspark via something like this:
```python
from pyspark.sql.types import *

schema = StructType().add("c1", IntegerType(), nullable=True, metadata={ 'parquet.field.id' : 1 })
```

Reply:
Done, moved to pyspark.